Mathlib/Algebra/Polynomial/Derivative.lean

```diff
@@ -577,6 +577,13 @@ theorem iterate_derivative_X_sub_pow_self (n : ℕ) (c : R) :
     derivative^[n] ((X - C c) ^ n) = n.factorial := by
   rw [iterate_derivative_X_sub_pow, n.sub_self, pow_zero, nsmul_one, n.descFactorial_self]
 
+theorem iterate_derivative_finset_sum (k : ℕ) (h : ℕ → ℕ) :
+    derivative^[k] (∑ m in Finset.range (k + 1), (h m * (-1 : R) ^ m) • X ^ (k + m)) =
+    ∑ m in Finset.range (k + 1), ((h m * (-1 : R) ^ m) • (derivative^[k] (X ^ (k + m))) : R[X]):= by
```

Comment:
This feels like a fairly odd statement. I'm wondering in particular what the `(-1 : R) ^ m` is for. Also there should be a space before `:=`

Reply:
this is used to deal with the n-derivative of something like (a - x) ^ n.

Reply:
So can you please add that result as a consequence?

Reply:
sure! I add  this result at another pr which defined shiftedlegendre polynomial.